exchange/index.go

```diff
@@ -453,6 +476,145 @@ func (idx *Index) evict(size uint64) uint64 {
 	return evicted
 }
 
+func (idx *Index) tagForEviction(ref *DataRef) error {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	return idx.walkDAG(ref.PayloadCID, func(blk blocks.Block) error {
+		idx.evictionSet.Add(blk.Cid())
+		return nil
+	})
+}
+
+func (idx *Index) GC() {
```

Comment:
```suggestion
func (idx *Index) GC() error {
```
Let's raise the error